IDE Integration Updates & Doc Updates

C-SELLERS opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Description
<!--- Describe your changes in detail -->
- Update VS Code readme instructions, launch options, and tasks
- Update VS readme instructions
- Remove PyCharm documentation
- Point users to Lean CLI for local algorithm development in all readmes

For the readme I placed Lean CLI as the first option for algorithm developers since it is the most friendly, supported, and links to QC users web accounts. I linked directly to QC's docs regarding using the Lean CLI

Then for local installs I covered how to install required dependencies, this was covered before but has changed significantly with dotnet 5. I also covered how to use Lean, again borrowing some pieces from the previous documentation.

#### Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
Closes #5549 , Closes #5572

#### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Lot of things have changed recently with the move to dotnet 5 framework and the introduction of the Lean CLI we needed to update Lean documentation to reflect it. 

#### Requires Documentation Change
<!--- Please indicate if these changes will require updates to documentation, and if so, specify what changes are required -->
Docs updated

#### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

For the local dependency install I tested the instructions in a windows VM and was successful in debugging Lean.

#### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Non-functional change (xml comments/documentation/etc)

#### Checklist:
<!--- The following is a checklist of items that MUST be completed before a PR is accepted -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. <!--- If not applicable, please explain why -->
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

<!--- Template inspired by https://www.talater.com/open-source-templates/#/page/99 -->

C-SELLERS commented on .vscode/tasks.json:
```diff
@@ -38,33 +39,19 @@
         },
+
+        // Below tasks are no longer being supported
+        // Succeeded by Lean CLI
```
Was thinking we might just remove these all together?? Maybe even remove the scripts themselves since they are built to run lean with mono debugger not dotnet.

This also means we should address pycharm integration since it also makes use of these scripts.

C-SELLERS replied:
Could possibly adapt them, but I believe all the important functionality was implemented in Lean CLI

Martin-Molinero reviewed (commented):
Cool!
Why are we removing `.idea` completely here?

C-SELLERS commented:
No longer needed it since we dropped the run_docker.sh/.bat configurations and [Lean/CLI automatically generates](https://www.quantconnect.com/docs/v2/lean-cli/user-guides/directory-structure#03-New-Python-projects) the required .idea files for the user to use PyCharm for algorithm development.

I adjusted and left the .vscode and .vs files because Lean development (not algos) can be done in these IDEs and this info is useful for opensource contributors. 

The only thing that I think it might be useful for would be a readme to route users to the Lean/CLI but I included a small bit about that in the main repo readme and it feels nice to drop a directory and cleanup the repo 😁
